### ISSUE TITLE
Styling badges - Loosing custom color

### ISSUE THREAD

**Issue by medicus07** [thomasloven/lovelace-card-mod#84]
My Home Assistant version: 0.115.1, problem in newer versions of HA, don't know since when... before it was working fine

My lovelace configuration method (GUI or yaml): YAML

What I am doing: 
I style my badges with card mod in the following way...

``` 
badges:
  - entity: binary_sensor.garage_status
    name: Garage
      style: |
        :host {
          --label-badge-blue: {% if is_state('binary_sensor.garage_status', 'off') %} green {% else %} red {% endif %};
        }
```

What I expected to happen: 
Changing color of the badge between green and red...

What happened instead: 
When refreshing webpage or app, there is no color in styled batches anymore. After clicking or touching screen, push a button or something else, the color appears again. In older versions of HA it worked just fine without this bug.

Screenshots:
<img width="340" alt="Bildschirmfoto 2020-11-22 um 08 53 56" src="https://user-images.githubusercontent.com/24975901/99898252-4e996f80-2ca0-11eb-8550-64bddce7c691.png">
<img width="344" alt="Bildschirmfoto 2020-11-22 um 08 53 40" src="https://user-images.githubusercontent.com/24975901/99898255-51946000-2ca0-11eb-90e0-91a9eacba9ab.png">

**Comment by KTibow** [comment]
Does it work in 0.118? Are you sure you only need to style `label-badge-blue`?

**Comment by medicus07** [comment]
No, with 0.118 it is still the same problem. 
For the binary sensor changing only the standard blue color shoulder be correct.

**Comment by KTibow** [comment]
It could be the feature that auto-disconnects the event stream. I believe binary sensors toggle between `label-badge-blue` and `label-badge-red`.

**Comment by medicus07** [comment]
Any chance to solve the problem?

**Comment by KTibow** [comment]
Can you give a reproducible config?

**Comment by medicus07** [comment]
in configuration.yaml I use the following:

```
frontend:
  themes: !include_dir_merge_named themes
  extra_module_url:
    - /local/custom-ui/custom-ui.js

lovelace:
  mode: yaml

homeassistant:
  customize:
    binary_sensor.garage_status:
      device_class: garage_door
      unit_of_measurement: 'Status'

binary_sensor:
  - platform: mqtt
    state_topic: "homee.garage"
    name: "Garage Status"
```

in ui-lovelace.yaml I use the following:

```
resources:
  - url: /hacsfiles/lovelace-card-mod/card-mod.js
    type: module

views:
  - badges:
      - entity: binary_sensor.garage_status
        name: Garage
        style: |
          :host {
            --label-badge-blue: {% if is_state('binary_sensor.garage_status', 'off') %} green {% else %} red {% endif %};
          }
```

**Comment by pfinnigan** [comment]
My card overrides stopped working in v118.2 when I upgraded from 117.? Where they worked. Tried the entity card color red example from this site on another card. Same issue, but that is new in 118.2

**Comment by KTibow** [comment]
Don't spam please.

**Comment by KTibow** [comment]
From what I can tell the template isn't rendering. I don't know how to solve this, but you can fallback like this:
```
views:
  - badges:
      - entity: binary_sensor.garage_status
        name: Garage
        style: |
          :host {
            --label-badge-blue: #039be5;
            --label-badge-blue: {% if is_state('binary_sensor.garage_status', 'off') %} green {% else %} red {% endif %};
          }
```

**Comment by medicus07** [comment]
Thank you agin... unfortunately the problem still exists.
So I did some testing around and could localize the problem with Safari and the HA iOS App (also Safari based).

With Chrome it works fine without any issue.
So it seems that Safari in newer versions uses another interpretation of CSS variables.

**Comment by Mariusthvdb** [comment]
think I am experiencing the same, using this badge styling:
```
using this:

badges:
  - entity: person.marijn
    <<: &badge
      style:
        ha-state-label-badge:
          $:
            ha-label-badge:
              $: |
                .label-badge .label span{
                  {% if is_state(config.entity,'home') %} color: gold;
                  {% else %} color: black;
                  {% endif %}
                }
          .: |
            :host {
              {% set zones = states.zone|map(attribute='name')|list %}
              {% if is_state(config.entity,'home') %} --label-badge-red: green;
              {% elif 'bij' in states(config.entity) %} --label-badge-red: gold;
              {% elif states(config.entity) in ['moving','driving'] or
                      states(config.entity) in zones %} --label-badge-red: mediumslateblue
              {% else %} --label-badge-red: dimgrey;
              {% endif %}
            }

Yaml anchor is for the other badges using this styling too. Of course ive tested without the anchor definition, but that doesn't make a difference, which would have been surprising, since it is perfect yaml and that is fully supported throughout HA configs.

Templates are correct (work on other entities in regular cards too) and make the styling appear, but only after sometime. 

If it does, a mere page reload makes the badges use their default unstyled colors, only to have the style mod kick back in after some time again. Can't say it has to do with any state change.

This is in HA 2021.1.4 meanwhile.

**Comment by thomasloven** [comment]
If it's an issue in safari but not in chrome it may be a priority issue. Try adding `!important` to the end of the style, maybe?

**Comment by farao780** [comment]
I use mainly firefox and after refreshing browser the custom-color is gone. Same thing in chrome.
Funny fact is that i enter in one card and just delete a blank space and press save, colourised icons appear again. If I try to do the same for second card, then colours dissapear from first card.

**Comment by thomasloven** [comment]
Not convinced this is the same issue. Please open a new issue if you still have problems, @farao780.
